### PR TITLE
Fixed RSA key pair generation method

### PR DESCRIPTION
**Pull request by proteye**
Fixed RSA key pair generation method.

### CODE REVIEW

**Comment by shutup** [comment]
i use this code in flutter,and it is run,thx!

File: lib/key_generators/rsa_key_generator.dart [hunk 1]
```diff
@@ -80,32 +80,32 @@ class RSAKeyGenerator implements KeyGenerator {
       for ( ; ; ) {
         q = generateProbablePrime(qbitlength, 1, _random);
 
-        if ((q - p).abs().bitLength() < mindiffbits) {
+        if ((q - p).abs().bitLength < mindiffbits) {
           continue;
         }
 
-        if (q.mod(e) == 1) {
```
Comment (stevenroose) [inline lib/key_generators/rsa_key_generator.dart:87]: Changing `mod` for `modInverse`, that seems not correct.. I think the `BigInt` method has the `%` operator implemented. So I think you can do `q % e == BigInt.one`.

File: lib/key_generators/rsa_key_generator.dart [hunk 2]
```diff
@@ -42,7 +42,7 @@ class RSAKeyGenerator implements KeyGenerator {
     }
   }
 
-  AsymmetricKeyPair generateKeyPair() {
```
Comment (stevenroose) [inline lib/key_generators/rsa_key_generator.dart:45]: Why a generic method?

**Review by stevenroose — CHANGES_REQUESTED** [review]

**Comment by stevenroose** [comment]
This commit should have been included in https://github.com/PointyCastle/pointycastle/pull/148. Could you, @proteye or @yshrsmz confirm this? Then this can be closed.

Thanks nonetheless!

**Comment by yshrsmz** [comment]
yes, I had included commits from this PR(and requested changes also), as these were needed to make the tests passed.

**Comment by stevenroose** [comment]
K, then I'm closing this. Thanks again!

**Comment by proteye** [comment]
Ok. Thank you!

**Pull request closed without merging**